packages/picasso/src/utils/use-body-scroll-lock.ts

```diff
@@ -0,0 +1,62 @@
+import { isBrowser } from '@toptal/picasso-shared'
+import { useEffect, useRef } from 'react'
+
+const layers: number[] = [];
+let scrollLock: { prevBodyOverflow: string } | undefined = undefined;
+
+export const useBodyScrollLock = (isLocked: boolean) => {
+  const layerId = useRef(generateLayerId())
```

Comment:
nit: just a small concern about `generateLayerId` being called every time a component using this hook is rendered. Memoization could help I think, but its probably not that big of deal. I'll leave it up to you

Reply:
I never want this value to change and do not want it to affect lifecycle of the component, so IMO useRef is the best fit for that case. Not sure how exactly and what for I'd combine memoization with useRef...

Reply:
Something like this was my idea. The generated id without memoization can grow to high numbers, but it's probably not a big deal. Feel free to resolve this conversation 👌 
```suggestion
  const id = useMemo(generateLayerId, [])
  const layerId = useRef(id)
```

Reply:
I feel like Ondra's concern is valid, we are incrementing the counter on every re-render, while I don't think this will cause any starvation on the counter (`Number.MAX_SAFE_INTEGER` is pretty big), I think of it as highly inefficient. The solution of `useMemo` that Ondra proposed would fix this behavior

Reply:
In fact, we might not even need a ref here, just the memoized number should be enough
```suggestion
  const layerId = useMemo(generateLayerId, [])
```
And later we can use it directly
```js
const lockedLayerId = layerId.current;
// Becomes
const lockedLayerId = layerId;
```

Reply:
Updated

Reply:
We actually have the same pattern used in [Modal.tsx#L134](https://github.com/toptal/picasso/blob/master/packages/picasso/src/Modal/Modal.tsx#L134) and I assume it was never an issue.

Reply:
We could hide the implementation behind our own `useId` hook, but I think is better to wait for Picasso moving on supporting [React 18 and beyond](https://beta.reactjs.org/reference/react/useId)